### PR TITLE
Bugfix port allowcategory

### PR DESCRIPTION
**Pull request by kumarnaresh**
Cargo Port cross validation.
Ready for review

### CODE REVIEW

**Comment by kumarnaresh** [comment]
Ready for review!

**Comment by karishmadft** [comment]
Why is this branch called bugfix-x? It was a feature 🤣

**Comment by kumarnaresh** [comment]
Nope!! lol 

I believe the branch is "bugfix-port-allowcategory"

**Comment by kumarnaresh** [comment]
Please ignore these two lines:
**private readonly uint CrudeOilCode = 12; 
private readonly uint OilProductsCode = 13;** 

These are obsolete and will be removed next time I work on this.

File: PortFreight.Web/Pages/Msd1/CargoDetails.cshtml.cs [hunk 1]
```diff
@@ -206,13 +216,50 @@ private void ManualModelValidation()
...
                     }
-                    
+
+                }
+
+                var categoryCode = _context.CargoCategory.AsQueryable().AsTracking()
+                    .Where(cc => cc.Description == CargoItem.Category).Select(c => c.CategoryCode).SingleOrDefault();
+
+                if (categoryCode == (int)PortCategory.CrudeOilCode)
+                {
```
Comment (karishmadft) [inline PortFreight.Web/Pages/Msd1/CargoDetails.cshtml.cs]: Although both 12 & 13 will have `grossWeightVisible`, this check isn't directly linked to that, so I don't think it should be within the `if (grossWeightVisible)` statement.

File: PortFreight.Web/Pages/Msd1/CargoDetails.cshtml.cs [hunk 2]
```diff
@@ -206,13 +216,50 @@ private void ManualModelValidation()
...
+                else if (categoryCode == (int)PortCategory.OilProductsCode)
+                {
+                    if (localMSD1.IsInbound == true && GetCountryCode(localMSD1.AssociatedPort).Equals("GB"))
+                    {
+                        var allowedCat = _context.PortCargoOil.AsQueryable().AsNoTracking()
+                       .Where(a => a.Locode == GetPortCode(localMSD1.AssociatedPort))
+                       .Select(a => a.AllowCategory13Outward).SingleOrDefault();
+                        if (!allowedCat)
+                        {
+                            ModelState.AddModelError("CargoItem.Category", "Cargo Category is not valid");
```
Comment (karishmadft) [inline PortFreight.Web/Pages/Msd1/CargoDetails.cshtml.cs]: More specific error would be better, 'Crude Oil cannot be carried through Port X'

Comment (kumarnaresh) [reply PortFreight.Web/Pages/Msd1/CargoDetails.cshtml.cs]: Al sounds good to me , let have a chat tomorrow! It will be quick though.

**Comment by kumarnaresh** [comment]
Ready to review ( unit test not done yet)

**Comment by kumarnaresh** [comment]
Ready for review - cargo port validation and unit tests
 .. code refactor to tidy-up etc.

**Pull request merged**